storybookjs/storybook
# CLI: Pass package manager to postinstall

## What I did
- Pass the name of the package manager as an option to the postinstall script
- Create a postinstall script for addon-themes
- Don't add `^` to version if the specified version isn't semver (Example : `next`, `canary`)

## Checklist for Contributors

### Testing
#### Manual testing

1. Open a sandbox of your choosing
2. `yarn add -D tailwindcss postcss autoprefixer`
3. `npx storybook@canary add @storybook/addon-themes@canary`
4. `@storybook/auto-config` should be run to configure tailwind themes

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-23913-sha-45573ae9`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-23913-sha-45573ae9). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-23913-sha-45573ae9`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-23913-sha-45573ae9) |
| **Triggered by** | @ndelangen |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`shaun/pass-package-manager`](https://github.com/storybookjs/storybook/tree/shaun/pass-package-manager) |
| **Commit** | [`45573ae9`](https://github.com/storybookjs/storybook/commit/45573ae9c7faaac835403dcce2e1605a164ede39) |
| **Datetime** | Wed Aug 23 14:35:40 UTC 2023 (`1692801340`) |
| **Workflow run** | [5952768837](https://github.com/storybookjs/storybook/actions/runs/5952768837) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=23913`_
</details>
